src/main.rs

```diff
@@ -65,7 +65,64 @@ enum Phrase {
 	Comment(String),
 }
 
-fn parse_whitespace(s: &mut Peekable<impl Iterator<Item = char>>) -> miette::Result<()> {
+struct Location {
+	row: isize,
+	col: isize,
+}
+
+impl Location {
+	pub fn next_col(&mut self) {
+		self.col += 1;
+	}
+
+	pub fn next_row(&mut self) {
+		self.row += 1;
+		self.col = 0;
+	}
+
+	pub fn default() -> Self {
+		Self { row: 0, col: 0 }
```

Comment:
small nits: VSCode starts at (1, 1) rather than (0, 0). I think the column varies from editor to editor (sometimes 1, sometimes 0) but the first line is pretty much always 1 unless you tweak some settings.

also you should move this function to an `impl Default for Location` block :)